Tests/SemanticTests/operand_types.flint

```diff
@@ -0,0 +1,27 @@
+// RUN: %flintc %s --verify
+
+contract Test {
+  var a: Int = 0
+  var b: Bool = false
+  var c: Address = 0x0000000000000000000000000000000000000000
+  var d: D = D()
+}
+
+Test :: (any) {
+  public init() {
+    a = 1
+    a = false // expected-error {{Incompatible assignment between values of type 'Int' and 'Bool'}}
+    a = b // expected-error {{Incompatible assignment between values of type 'Int' and 'Bool'}}
+    a = c // expected-error {{Incompatible assignment between values of type 'Int' and 'Address'}}
+    a = d // expected-error {{Incompatible assignment between values of type 'Int' and 'D'}}
+    d = 0 // expected-error {{Incompatible assignment between values of type 'D' and 'Int'}}
+    (a < 1)
+    (a < b) // expected-error {{Incompatible use of operator '<' on values of types 'Int' and 'Bool'}}
+    (a > c) // expected-error {{Incompatible use of operator '>' on values of types 'Int' and 'Address'}}
```

Comment:
Mind explaining to me why these are bracketed?

Reply:
I thought the lack of bracketing might have been the reason for a crash when debugging, and forgot to remove them after it turned out not to be. Fixed now.